RecoPixelVertexing/PixelTriplets/plugins/CAHitQuadrupletGeneratorGPU.cc

```diff
@@ -0,0 +1,460 @@
+        auto hitId = i * maxNumberOfDoublets * 2 + 2 * l;
+        assert(maxNumberOfDoublets >= hitDoublets[index][i]->size());
+        h_indices[hitId] = hitDoublets[index][i]->innerHitId(l);
+        h_indices[hitId + 1] = hitDoublets[index][i]->outerHitId(l);
+      }
+    }
+
+    for (unsigned int j = 0; j < numberOfLayers; ++j) {
+      for (unsigned int l = 0; l < h_layers[j].size; ++l) {
+        auto hitId = h_layers[j].layerId * maxNumberOfHits + l;
+        assert(h_x[hitId] != 0);
+      }
+    }
+
+    for (unsigned int j = 0; j < numberOfLayerPairs; ++j) {
+      tmp_layerDoublets[j] = h_doublets[j];
+      tmp_layerDoublets[j].indices = &d_indices[j * maxNumberOfDoublets * 2];
+      cudaMemcpyAsync(&d_indices[j * maxNumberOfDoublets * 2],
+                      &h_indices[j * maxNumberOfDoublets * 2],
+                      tmp_layerDoublets[j].size * 2 * sizeof(int),
+                      cudaMemcpyHostToDevice, cudaStream_);
+    }
+
+    for (unsigned int j = 0; j < numberOfLayers; ++j) {
+      tmp_layers[j] = h_layers[j];
+      tmp_layers[j].x = &d_x[maxNumberOfHits * j];
+
+      cudaMemcpyAsync(&d_x[maxNumberOfHits * j], &h_x[j * maxNumberOfHits],
+                      tmp_layers[j].size * sizeof(float),
+                      cudaMemcpyHostToDevice, cudaStream_);
+
+      tmp_layers[j].y = &d_y[maxNumberOfHits * j];
+      cudaMemcpyAsync(&d_y[maxNumberOfHits * j], &h_y[j * maxNumberOfHits],
+                      tmp_layers[j].size * sizeof(float),
+                      cudaMemcpyHostToDevice, cudaStream_);
+
+      tmp_layers[j].z = &d_z[maxNumberOfHits * j];
+
+      cudaMemcpyAsync(&d_z[maxNumberOfHits * j], &h_z[j * maxNumberOfHits],
+                      tmp_layers[j].size * sizeof(float),
+                      cudaMemcpyHostToDevice, cudaStream_);
+    }
+
+    cudaMemcpyAsync(d_rootLayerPairs, h_rootLayerPairs,
+                    numberOfRootLayerPairs * sizeof(unsigned int),
+                    cudaMemcpyHostToDevice, cudaStream_);
+    cudaMemcpyAsync(d_doublets, tmp_layerDoublets,
+                    numberOfLayerPairs * sizeof(GPULayerDoublets),
+                    cudaMemcpyHostToDevice, cudaStream_);
+    cudaMemcpyAsync(d_layers, tmp_layers, numberOfLayers * sizeof(GPULayerHits),
+                    cudaMemcpyHostToDevice, cudaStream_);
+
+    launchKernels(region, index);
+}
+}
+
+void CAHitQuadrupletGeneratorGPU::fillResults(
+    const IntermediateHitDoublets &regionDoublets,
+    std::vector<OrderedHitSeeds> &result, const edm::EventSetup &es,
+    const SeedingLayerSetsHits &layers, const cudaStream_t &cudaStream)
+{
+    int index = 0;
+
+    for (const auto &regionLayerPairs : regionDoublets) {
+      const TrackingRegion &region = regionLayerPairs.region();
+      auto foundQuads = fetchKernelResult(index);
+      std::cout << foundQuads.size() << " found quads" << std::endl;
+    unsigned int numberOfFoundQuadruplets = foundQuads.size();
```

Comment:
Indentation becomes inconsistent at this line. But I'd rather re-indent the lines above than those below.